### PR TITLE
Fix class names mismatch between command result and remaining parts of GUI

### LINKED ISSUE #265: Use `PREFIX_MODULE_CLASS` instead of `c/` directly in focus command.

### LINKED ISSUE #312: [PE-D][Tester E] Student class tagnames not the same as class name
**Description**

It appears that the tagname of the student is dependent on the capitalisation used by the user in the `assign` command rather than the actual class name the student is being assigned to.

I think it would be perhaps better to stick to either displaying class names in using the capitalisation the class was created with. After all, they all refer to the same class, and so, should have the same name (including capitalisation).

**Steps to reproduce**
1. Assign a student (who was previously unassigned) to the class `assign 1 c/CS1101S`
2. Now assign another student, but change the capitalisation of the class specified `assign 2 c/cs1101s`
3. Notice that though the 2 students are assigned to the same class (CS1101S), they have different tagnames.

**Screenshot**

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: chloeelim/ped#4

### LINKED ISSUE #314: [PE-D][Tester E] Inconsistent naming of product
In the UG, the product name is referred to as "TA-Assist" sometimes (the first word of the UG, in the Quick Start section, and in the command formatting convention callout box [second last paragraph]), and "TA Assist" in other places (rest of the documentation and also the window header of the actual product).

Perhaps standardising the product name would be good?

**Screenshots**

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: chloeelim/ped#6

Name updated in:
- [x] UG
- [x] DG

### PR DESCRIPTION
**Pull request by njxue**
Resolves class names in #255 but not session names.
Simplest fix is to just make class names all CAPS but Im not sure if we are allowed to do that in v1.4 (if they are very strict), so I didn't resort to that.
Also fixed some minor UG bugs raised during PE-D

### CODE REVIEW

File: src/main/java/seedu/taassist/model/TaAssist.java [hunk 1]
```diff
@@ -162,7 +165,14 @@ public void removeModuleClass(ModuleClass moduleClass) {
         setStudents(updatedStudents);
     }
 
-
+    public ModuleClass getModuleClassWithSameName(ModuleClass moduleClass) throws ModuleClassNotFoundException {
+        requireNonNull(moduleClass);
+        Optional<ModuleClass> foundModuleClass = moduleClasses.findElement(moduleClass);
+        if (foundModuleClass.isEmpty()) {
```
Comment (Berted) [inline src/main/java/seedu/taassist/model/TaAssist.java]: Perhaps consider using the method `Optional::orElseThrow` to simplify this code section?

File: src/test/java/seedu/taassist/logic/commands/AssignCommandTest.java [hunk 2]
```diff
@@ -37,7 +40,7 @@ public void execute_noModuleClass_throwsCommandException() {
     }
 
     @Test
-    public void execute_indexOutOfRange_throwsCommandException() {
+    public void execute_indexOutOfRange_throwsCommandException() throws Exception {
```
Comment (Berted) [inline src/test/java/seedu/taassist/logic/commands/AssignCommandTest.java]: Any reason why this method throws an `Exception`?

File: src/test/java/seedu/taassist/logic/commands/AssignCommandTest.java [hunk 3]
```diff
@@ -135,19 +121,27 @@ public ObservableList<Student> getFilteredStudentList() {
...
+            requireNonNull(moduleClass);
+            return moduleClass;
+        }
     }
 
     /**
      * Model stub with multiple filtered students. Always says it has a module.
      */
-    private class ModelStubWithMultipleStudents extends ModelStub {
+    private static class ModelStubWithMultipleStudents extends ModelStub {
```
Comment (Berted) [inline src/test/java/seedu/taassist/logic/commands/AssignCommandTest.java:144]: Good catch!

**Review by Berted — CHANGES_REQUESTED** [review]
Nice work! Albeit I have a few nits regarding some sections of the code, but overall, good work!

File: src/test/java/seedu/taassist/logic/commands/AssignCommandTest.java (hunk 2 above)
Comment (njxue) [reply src/test/java/seedu/taassist/logic/commands/AssignCommandTest.java]: oops forgot to delete

**Review by Berted — APPROVED** [review]
LGTM!

**Pull request merged**